pkg/api/common/const.go

```diff
@@ -103,8 +103,6 @@ const (
 	ContainerMonitoringAddonName = "container-monitoring"
 	// IPMASQAgentAddonName is the name of the ip masq agent addon
 	IPMASQAgentAddonName = "ip-masq-agent"
-	// AzureCNINetworkMonitorAddonName is the name of the Azure CNI networkmonitor addon
-	AzureCNINetworkMonitorAddonName = "azure-cni-networkmonitor"
```

Comment:
If we remove the const definition, will cluster templates which have `enabled: True` for the `azure-cni-networkmonitor` addon now raise an error? I don't remember how this would behave... And if it does error, are we ok with that instead of having a "deprecated addon was ignored"-type message?

Reply:
Good call-out, I'll add some deprecation foo for clusters transitioning during an upgrade to the new "no Azure CNI networkmonitor addon exists" reality.

Reply:
Made some changes, check out the latest commit